perceval/backends/git.py

```diff
@@ -642,11 +654,16 @@ def log(self, from_date=None, encoding='utf-8'):
         """
 
         cmd_log = ['git', 'log', '--raw', '--numstat', '--pretty=fuller',
-                   '--decorate=full', '--all', '--reverse', '--topo-order',
-                   '--parents', '-M', '-C', '-c', '--remotes=origin']
+                   '--decorate=full', '--reverse', '--topo-order',
+                   '--parents', '-M', '-C', '-c']
         if from_date:
             dt = from_date.strftime("%Y-%m-%d %H:%M:%S %z")
             cmd_log.append('--since=' + dt)
+        if branches:
+            branches = ['remotes/origin/' + branch for branch in branches]
```

Comment:
As I can understand, when `branches` is an empty list or `None` the behaviour is to fetch every commit. Is this the semantic we want for an empty list?